### PR TITLE
Simplify NavTree code

### PR DESCRIPTION
**Pull request by marrony**
Still need to refactor `NavDBTree`. That component is huge and hard to refactor

### CODE REVIEW

**Comment by erickpintor** [comment]
Does this contain any substantial fix? I'd prefer to hold on refactorings until after the release.

**Comment by marrony** [comment]
No, it's only a refactoring

**Comment by erickpintor** [comment]
Let's hold on until we release the last bug fixes then.

**Comment by jchris** [comment]
I can see how this will help with #96 -- I'll rebase with master and test locally. We should probably wait to merge until after we deploy.

**Comment by jchris** [comment]
I rebased this to master, and it looks good / feels good. Except I need help fixing one failing test. I think we need to stub classes and indexes into the login store, because of how the dispatches are interlinked. So maybe the right arguments to `createTestStore` in `login.test.js`?

Or else we can link the dispatch calls somehow...

File: src/app/clients.js [hunk 1]
```diff
@@ -9,13 +12,21 @@ export function updateClients(rootClient, splat) {
...
       type: Actions.UPDATE,
       rootClient: rootClient,
-      scopedServerClient: clientForSubDB(rootClient, splat, "server"),
-      scopedAdminClient: clientForSubDB(rootClient, splat, "admin"),
+      scopedServerClient: scopedServerClient,
+      scopedAdminClient: scopedAdminClient,
       splat: splat
     })
+
+    return Promise.all([
```
Comment (marrony) [inline src/app/clients.js]: I think we can move this out of here, and do right after we update clients in that condition
```javascript
if(getState().currentUser) {
   const rootClient = getState().currentUser.client
   dispatch(updateClients(rootClient, splat))
```

Comment (marrony) [inline src/app/clients.js]: That way we don't have to attach the login with fetching data and it will not break the tests

Comment (jchris) [inline src/app/clients.js]: good idea, lemme see

**Comment by jchris** [comment]
I was able to move the class and index concerns up to the app level and the tests started passing.

**Comment by jchris** [comment]
I'm open to merging this in the morning, so we have time to work out any bugs before we deploy.

**Pull request merged**